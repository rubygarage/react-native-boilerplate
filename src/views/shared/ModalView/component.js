import React from 'react';
import PropTypes from 'prop-types';
import SafeAreaView from 'react-native-safe-area-view';

import * as S from './styled';

const ModalView = ({
  isVisible, onClose, children, onModalShow, onModalHide,
}) => (
  <S.Modal
    coverScreen={false}
    isVisible={isVisible}
    onBackButtonPress={onClose}
    onBackdropPress={onClose}
    onModalShow={onModalShow}
    onModalHide={onModalHide}
  >
    <S.Container>
      <SafeAreaView>
        {children}
      </SafeAreaView>
    </S.Container>
  </S.Modal>
);

ModalView.propTypes = {
  children: PropTypes.node.isRequired,
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func,
  onModalShow: PropTypes.func,
  onModalHide: PropTypes.func,
};

ModalView.defaultProps = {
  onModalShow: undefined,
  onModalHide: undefined,
  onClose: undefined,
};

export default ModalView;
