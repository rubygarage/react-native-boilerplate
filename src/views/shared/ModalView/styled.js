import styled from 'styled-components/native';
import ModalView from 'react-native-modal';

const Modal = styled(ModalView)`
  flex-grow: 1;
  justify-content: flex-end;
  margin: 0;
`;

const Overlay = styled.View`
  flex-grow: 1;
  justify-content: flex-end;
`;

const Container = styled.View`
  background-color: ${({ theme }) => theme.colors.white};
  border-top-left-radius: 4;
  border-top-right-radius: 4;
`;

export { Modal, Overlay, Container };
