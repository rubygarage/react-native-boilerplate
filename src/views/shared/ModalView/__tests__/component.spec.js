import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';

import ModalView from '../component';

const defaultProps = {
  isVisible: true,
  onClose: jest.fn(),
};

describe('ModalView component', () => {
  const component = shallow(<ModalView {...defaultProps}><View /></ModalView>);
  it('renders correctly visible', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correctly invisible', () => {
    const props = { ...defaultProps, isVisible: false };
    component.setProps(props);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when is not cancelable', () => {
    const props = { ...defaultProps, onClose: undefined };
    component.setProps(props);
    expect(component).toMatchSnapshot();
  });
});
