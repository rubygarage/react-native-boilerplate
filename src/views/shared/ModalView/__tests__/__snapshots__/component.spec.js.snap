// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView component renders correctly invisible 1`] = `
<Styled(ReactNativeModal)
  coverScreen={false}
  isVisible={false}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
>
  <Styled(View)>
    <SafeAreaView>
      <View />
    </SafeAreaView>
  </Styled(View)>
</Styled(ReactNativeModal)>
`;

exports[`ModalView component renders correctly visible 1`] = `
<Styled(ReactNativeModal)
  coverScreen={false}
  isVisible={true}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
>
  <Styled(View)>
    <SafeAreaView>
      <View />
    </SafeAreaView>
  </Styled(View)>
</Styled(ReactNativeModal)>
`;

exports[`ModalView component renders correctly when is not cancelable 1`] = `
<Styled(ReactNativeModal)
  coverScreen={false}
  isVisible={true}
>
  <Styled(View)>
    <SafeAreaView>
      <View />
    </SafeAreaView>
  </Styled(View)>
</Styled(ReactNativeModal)>
`;
